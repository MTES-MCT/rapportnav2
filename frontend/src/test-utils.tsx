import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
