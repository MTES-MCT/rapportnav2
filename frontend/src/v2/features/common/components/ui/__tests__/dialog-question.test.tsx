import { vi } from 'vitest'
import { fireEvent, render } from '../../../../../../test-utils'
import DialogQuestion from '../dialog-question'

const onSubmit = vi.fn()
describe('DialogQuestion', () => {
  beforeEach(() => {
    onSubmit.mockClear()
  })
  it('should match the snapshot', () => {
    const wrapper = render(
      <DialogQuestion
        type="danger"
        title="Suppression de la mission"
        question="Voulez vous vraiment supprimer cette mission?"
        onSubmit={onSubmit}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should trigger on submit with true', () => {
    const wrapper = render(<DialogQuestion type="danger" title="" question="" onSubmit={onSubmit} />)
    const confirmButton = wrapper.getByTestId('dialog-question-confirm-button')
    fireEvent.click(confirmButton)
    expect(onSubmit).toHaveBeenCalledTimes(1)
    expect(onSubmit).toHaveBeenCalledWith(true)
  })

  it('should trigger on submit with false', () => {
    const wrapper = render(<DialogQuestion type="danger" title="" question="" onSubmit={onSubmit} />)
    const confirmButton = wrapper.getByTestId('dialog-question-cancel-button')
    fireEvent.click(confirmButton)
    expect(onSubmit).toHaveBeenCalledTimes(1)
    expect(onSubmit).toHaveBeenCalledWith(false)
  })
})
