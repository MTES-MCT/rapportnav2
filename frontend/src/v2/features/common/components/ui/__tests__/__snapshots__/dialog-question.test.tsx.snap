// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogQuestion > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ggWZvA jdgKwO Component-Dialog"
      >
        <div
          class="sc-dTvVRJ eLHUDB"
        />
        <div
          class="sc-jwTyAe lernAL"
        >
          <h4
            class="sc-fhHczv hDFPnI"
          >
            <div
              class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-space-between"
              style="padding-left: 14px; padding-right: 24px;"
            >
              <div
                class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
                style="font-size: 16px;"
              >
                Suppression de la mission
              </div>
              <div
                class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
              >
                <button
                  class="sc-dNdcvo bDxZrn Element-IconButton"
                  data-testid="close-dialog-question"
                  role="dialog-question"
                  type="button"
                >
                  <span
                    class="sc-hjsuWn hZNHWy Element-IconBox"
                  >
                    <svg
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <g
                        transform="translate(160 -82)"
                      >
                        <g
                          fill="none"
                          stroke-miterlimit="10"
                        >
                          <path
                            d="M-142,85.415-143.415,84-150,90.585-156.585,84-158,85.415-151.415,92-158,98.585l1.415,1.415L-150,93.415l6.585,6.585L-142,98.585-148.585,92Z"
                            stroke="none"
                          />
                          <path
                            d="M -156.5853271484375 84 L -150 90.5853271484375 L -143.4146728515625 84 L -142 85.4146728515625 L -148.5853271484375 92 L -142 98.5853271484375 L -143.4146728515625 100 L -150 93.4146728515625 L -156.5853271484375 100 L -158 98.5853271484375 L -151.4146728515625 92 L -158 85.4146728515625 L -156.5853271484375 84 Z"
                            fill="currentColor"
                            stroke="none"
                          />
                        </g>
                        <rect
                          fill="none"
                          height="20"
                          transform="translate(-160 82)"
                          width="20"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </h4>
          <div
            class="sc-hvigdm bPqtmR"
          >
            <div
              class="rs-stack"
              style="align-items: flex-start; justify-content: center; flex-direction: row; font-weight: bold; color: rgb(225, 0, 15);"
            >
              <div
                class="rs-stack-item"
              >
                <span
                  class="sc-hjsuWn eVTFbL Element-IconBox"
                >
                  <svg
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <g
                      transform="translate(200 -82)"
                    >
                      <g
                        fill="none"
                        stroke-miterlimit="10"
                      >
                        <path
                          d="M-190,83a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-190,83Zm0,16a7,7,0,0,1-7-7,7,7,0,0,1,7-7,7,7,0,0,1,7,7A7,7,0,0,1-190,99Z"
                          stroke="none"
                        />
                        <path
                          d="M -190 83 C -185.0290069580078 83 -181 87.02899932861328 -181 92 C -181 96.97100067138672 -185.0290069580078 101 -190 101 C -194.9709930419922 101 -199 96.97100067138672 -199 92 C -199 87.02899932861328 -194.9709930419922 83 -190 83 Z M -190 99 C -186.1340026855469 99 -183 95.86599731445313 -183 92 C -183 88.13400268554688 -186.1340026855469 85 -190 85 C -193.8659973144531 85 -197 88.13400268554688 -197 92 C -197 95.86599731445313 -193.8659973144531 99 -190 99 Z"
                          fill="currentColor"
                          stroke="none"
                        />
                      </g>
                      <g
                        fill="currentColor"
                        stroke="currentColor"
                        stroke-miterlimit="10"
                        stroke-width="2"
                        transform="translate(-189 93) rotate(180)"
                      >
                        <rect
                          height="5"
                          stroke="none"
                          width="2"
                        />
                      </g>
                      <g
                        fill="currentColor"
                        stroke="currentColor"
                        stroke-miterlimit="10"
                        stroke-width="2"
                        transform="translate(-189 96) rotate(180)"
                      >
                        <rect
                          height="2"
                          stroke="none"
                          width="2"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
              <div
                class="rs-stack-item"
                style="margin-left: 4px;"
              >
                Voulez vous vraiment supprimer cette mission?
              </div>
            </div>
          </div>
          <div
            class="sc-bRKDuR cyuotf"
          >
            <button
              class="sc-eknHtZ sc-jYczwO eIVVCb jxeylY Element-Button"
              data-testid="dialog-question-cancel-button"
              type="button"
            >
              <span
                class="sc-pYNGo enBrtN"
              >
                Annuler
              </span>
            </button>
            <button
              class="sc-eknHtZ sc-cCVJLD eIVVCb eUzjDK Element-Button"
              data-testid="dialog-question-confirm-button"
              type="button"
            >
              <span
                class="sc-pYNGo enBrtN"
              >
                Confirmer
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ggWZvA jdgKwO Component-Dialog"
    >
      <div
        class="sc-dTvVRJ eLHUDB"
      />
      <div
        class="sc-jwTyAe lernAL"
      >
        <h4
          class="sc-fhHczv hDFPnI"
        >
          <div
            class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-space-between"
            style="padding-left: 14px; padding-right: 24px;"
          >
            <div
              class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
              style="font-size: 16px;"
            >
              Suppression de la mission
            </div>
            <div
              class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
            >
              <button
                class="sc-dNdcvo bDxZrn Element-IconButton"
                data-testid="close-dialog-question"
                role="dialog-question"
                type="button"
              >
                <span
                  class="sc-hjsuWn hZNHWy Element-IconBox"
                >
                  <svg
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <g
                      transform="translate(160 -82)"
                    >
                      <g
                        fill="none"
                        stroke-miterlimit="10"
                      >
                        <path
                          d="M-142,85.415-143.415,84-150,90.585-156.585,84-158,85.415-151.415,92-158,98.585l1.415,1.415L-150,93.415l6.585,6.585L-142,98.585-148.585,92Z"
                          stroke="none"
                        />
                        <path
                          d="M -156.5853271484375 84 L -150 90.5853271484375 L -143.4146728515625 84 L -142 85.4146728515625 L -148.5853271484375 92 L -142 98.5853271484375 L -143.4146728515625 100 L -150 93.4146728515625 L -156.5853271484375 100 L -158 98.5853271484375 L -151.4146728515625 92 L -158 85.4146728515625 L -156.5853271484375 84 Z"
                          fill="currentColor"
                          stroke="none"
                        />
                      </g>
                      <rect
                        fill="none"
                        height="20"
                        transform="translate(-160 82)"
                        width="20"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </h4>
        <div
          class="sc-hvigdm bPqtmR"
        >
          <div
            class="rs-stack"
            style="align-items: flex-start; justify-content: center; flex-direction: row; font-weight: bold; color: rgb(225, 0, 15);"
          >
            <div
              class="rs-stack-item"
            >
              <span
                class="sc-hjsuWn eVTFbL Element-IconBox"
              >
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <g
                    transform="translate(200 -82)"
                  >
                    <g
                      fill="none"
                      stroke-miterlimit="10"
                    >
                      <path
                        d="M-190,83a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-190,83Zm0,16a7,7,0,0,1-7-7,7,7,0,0,1,7-7,7,7,0,0,1,7,7A7,7,0,0,1-190,99Z"
                        stroke="none"
                      />
                      <path
                        d="M -190 83 C -185.0290069580078 83 -181 87.02899932861328 -181 92 C -181 96.97100067138672 -185.0290069580078 101 -190 101 C -194.9709930419922 101 -199 96.97100067138672 -199 92 C -199 87.02899932861328 -194.9709930419922 83 -190 83 Z M -190 99 C -186.1340026855469 99 -183 95.86599731445313 -183 92 C -183 88.13400268554688 -186.1340026855469 85 -190 85 C -193.8659973144531 85 -197 88.13400268554688 -197 92 C -197 95.86599731445313 -193.8659973144531 99 -190 99 Z"
                        fill="currentColor"
                        stroke="none"
                      />
                    </g>
                    <g
                      fill="currentColor"
                      stroke="currentColor"
                      stroke-miterlimit="10"
                      stroke-width="2"
                      transform="translate(-189 93) rotate(180)"
                    >
                      <rect
                        height="5"
                        stroke="none"
                        width="2"
                      />
                    </g>
                    <g
                      fill="currentColor"
                      stroke="currentColor"
                      stroke-miterlimit="10"
                      stroke-width="2"
                      transform="translate(-189 96) rotate(180)"
                    >
                      <rect
                        height="2"
                        stroke="none"
                        width="2"
                      />
                    </g>
                  </g>
                </svg>
              </span>
            </div>
            <div
              class="rs-stack-item"
              style="margin-left: 4px;"
            >
              Voulez vous vraiment supprimer cette mission?
            </div>
          </div>
        </div>
        <div
          class="sc-bRKDuR cyuotf"
        >
          <button
            class="sc-eknHtZ sc-jYczwO eIVVCb jxeylY Element-Button"
            data-testid="dialog-question-cancel-button"
            type="button"
          >
            <span
              class="sc-pYNGo enBrtN"
            >
              Annuler
            </span>
          </button>
          <button
            class="sc-eknHtZ sc-cCVJLD eIVVCb eUzjDK Element-Button"
            data-testid="dialog-question-confirm-button"
            type="button"
          >
            <span
              class="sc-pYNGo enBrtN"
            >
              Confirmer
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
